symengine/complex.h

```diff
@@ -64,15 +64,14 @@ class Complex : public Number {
     static RCP<const Number> from_two_nums(const Number &re,
         const Number &im);
 
+    //! \return `true` if exactly `0`
+    virtual bool is_exact_zero() const { return false; }
     //! \return `true` if `0`
-    virtual bool is_zero() const { return ((this->real_ == 0) && (this->imaginary_ == 0)); }
+    virtual bool is_zero() const { return false; }
```

Comment:
I see. Can you put the check into an assert statement, that it is not zero? And add a comment that it cannot be zero, since it is in canonical form? That way it will be self documenting and also a bit safer in case something happened with the canonicalization.